Composer/packages/client/src/pages/design/DesignPage.tsx

```diff
@@ -313,17 +320,17 @@ const DesignPage: React.FC<RouteComponentProps<{ dialogId: string; projectId: st
     if (currentDialog.content) {
       setBreadcrumbs((prev) => {
         return prev.map((b) => {
-          const separatorIndex = b.key.indexOf('-');
-          const type = b.key.substr(0, separatorIndex);
-          const name = b.key.substr(separatorIndex + 1);
+          // eslint-disable-next-line security/detect-unsafe-regex
+          const regexp = new RegExp(/(?<type>[a-z]+)-(?<name>.*)/);
+          const { type, name } = b.key.match(regexp)?.groups || {};
```

Comment:
Seeing the linter need to be disabled here made me wonder if there were a simpler way to test this, and I think I have an idea how we could. This is an optional fix, but it will both get rid of the linter and make this a little faster.

Because we control the values of that Types enum and they aren't being exported anywhere, we could make the values a fixed length - say, single letters "D", "T", and "A" for dialog, trigger, and action (so a key would be `D-nameOfDialog`, for example). Then we'll always know the length of the type prefix and won't need any kind of searching or regex matching. The type would always be `b.key.charAt(0)` (or just `b.key[0]`), and the name would be `b.key.substr(2)`.

Reply:
Good idea, thanks